examples/cutting_stock_problem.rs

```diff
@@ -0,0 +1,179 @@
+/// to `\sum_t λ_{t} <= k` which is artificial anyway.
+///
+/// # Restricted Master Problem and Pricing Problem
+///
+/// To practically solve the master problem, we need to restrict the set of all patterns to a reasonably small set. The result is a
+/// restricted master problem. A reasonable set of patterns to start with is the set of all patterns that contain exactly one item.
+///
+/// The optimal solution to a restricted master problem is a feasible solution to the original problem. Taking an associated restricted dual
+/// solution, it may or may not be the case that the dual solution is feasible for the original problem. If it is, we would have proven
+/// optimality to the linear problem. If it is not, there is at least one dual constraint that is violated. The dual of the master problem
+/// is
+///
+/// ```
+///   \max \sum_i b_i π_i
+///     s.t. \sum_i π_i z_{i,t} <= 1    [λ_t]
+///          π_i >= 0
+/// ```
+///
+/// Note that the constraint corresponding to `λ_t` is violated iff its slack, known as its reduced cost
+/// `\bar{c}(λ_t) = 1 - \sum_i π_i z_{i,t}`, is negative. Of course, instead of checking for all `t`, we introduce the pricing problem
+///
+/// ```
+///   \min 1 - \sum_i π_i z_{i}
+///     s.t. \sum_i w_i z_{i} <= W
+///          z_{i} >= 0
+/// ```
+///
+// Following https://scipbook.readthedocs.io/en/latest/bpp.html and https://www.gerad.ca/fr/papers/G-2024-36.pdf
+fn main() {
+    let stock_length = 9;
+    let item_sizes = &[6.0, 5.0, 4.0, 2.0, 3.0, 7.0, 5.0, 8.0, 4.0, 5.0];
+    let demand = &[2, 3, 4, 4, 2, 2, 2, 2, 2, 1];
+
+    // Vector of cutting_patterns, initially populated with the trivial ones that contain exactly
+    // one item. cutting_patterns[i][j] indicates how often item j is in pattern i.
+    let initial_cutting_patterns: Vec<Vec<i32>> = (0..item_sizes.len()).map(|i|
+        (0..item_sizes.len()).map(|j| if i == j { 1 } else { 0 }).collect::<Vec<i32>>()
+    ).collect();
+
+    let mut main_problem = Model::default().minimize();
+
+    let cutting_pattern_vars: Vec<Variable> = initial_cutting_patterns.iter().enumerate().map(|(i, &ref _pattern)| {
+        let pattern = (0..10).map(|x| if x == i { "1" } else { "0" }).collect::<Vec<_>>().join("-");
+        main_problem.add_var(0.0, f64::INFINITY, 1.0, &format!("pattern_{pattern}"), VarType::Integer)
+    }).collect();
+
+    demand.iter().enumerate().for_each(|(i, &count)| {
+        let cons = main_problem.add_cons(
+            cutting_pattern_vars.iter().collect(),
+            &initial_cutting_patterns.iter().map(|pattern| pattern[i] as f64).collect::<Vec<f64>>(),
+            count as f64,
+            f64::INFINITY,
+            &format!("demand_for_item_{i}"));
+        main_problem.set_cons_modifiable(&cons, true);
+    });
+
+    let pricer = CSPPricer {
+        stock_length,
+        item_sizes,
+    };
+
+    main_problem.include_pricer("CSPPricer", "CSPPricer", 1, false, Box::new(pricer));
```

Comment:
here too, the new api would be more readable I think (assuming the pricer variable is renamed to p)
```suggestion
    main_problem.add(pricer(p).name("CSPPricer"));
```

Reply:
done